src/validators/dataclass.rs

```diff
@@ -476,10 +479,31 @@ impl Validator for DataclassValidator {
         let class = self.class.as_ref(py);
         if matches!(extra.mode, InputType::Python) && input.to_object(py).as_ref(py).is_instance(class)? {
             if self.revalidate.should_revalidate(input, class) {
-                let input = input.input_get_attr(intern!(py, "__dict__")).unwrap()?;
+                let mut validator_input = PyDict::new(py);
+                if self.slots {
+                    let slots = input
+                        .input_get_attr(intern!(py, "__slots__"))
+                        .unwrap()?
+                        .downcast::<PyTuple>()?;
+                    for key in slots.iter() {
+                        let key: &PyString = key.downcast()?;
+                        validator_input.set_item(key, input.input_get_attr(key).unwrap()?)?;
+                    }
+                } else {
+                    validator_input = input
+                        .input_get_attr(intern!(py, "__dict__"))
+                        .unwrap()?
+                        .downcast::<PyDict>()?;
+                }
```

Comment:
probably better to move this into a separate function, and return `&PyAny` as that's what `input` will need to be.